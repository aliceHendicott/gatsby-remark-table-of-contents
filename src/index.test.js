/* globals test describe  expect */
import remark from "remark";
import gatsbyTOC from ".";

const MD = [
  `
# Headlinge 1.0.0

## Table of Contents
`,

  `
## Headline 1.1.0

### Headline 1.1.1
#### Headline 1.1.1.0

## Headline 1.2.0
`,
];

const mdDefaultPrefs = () => `
${MD[0]}

\`\`\`toc
# This code block gets replaced with the TOC
\`\`\`

${MD[1]}
`;

const mdYamlPrefs = ({
  exclude,
  tight,
  ordered,
  fromHeading,
  toHeading,
  className,
}) => `
${MD[0]}

\`\`\`toc
# This code block gets replaced with the TOC
exclude: ${exclude || "NOTING"}
tight: ${tight ? "true" : "false"}
ordered: ${ordered ? "true" : "false"}
from-heading: ${fromHeading || 2}
to-heading: ${toHeading || 6}
class-name: "${className || "toc"}"
\`\`\`

${MD[1]}
`;

const toc = (config) => {
  return (markdownAST) => {
    gatsbyTOC(
      { markdownAST },
      {
        exclude: "NOTING",
        tight: false,
        fromHeading: 2,
        toHeading: 6,
        className: "toc",
        ...(config || {}),
      }
    );
  };
};

describe("TOC function", () => {
  test("Don't do TOC if not asked for it", async () => {
    const data = await remark().use(toc).process("# h1# h2");
    expect(data.contents.trim()).toEqual("# h1# h2");
  });

  test("illegal toc-definition should not throw an error but take defaults", async () => {
    const spy = jest.spyOn(console, "log").mockImplementation();

    const data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ exclude: "Table of Contents\n-a" }));
    expect(data.contents).toMatchSnapshot();

    expect(console.log).toBeCalled();

    spy.mockRestore();
  });

  test("exclude to exclude header", async () => {
    let data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ exclude: "Table of Contents" }));
    expect(data.contents).toMatchSnapshot();

    let dataPref = await remark()
      .use(() => toc({ exclude: "Table of Contents" }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);

    data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ exclude: "Headline 1.1.1" }));
    expect(data.contents).toMatchSnapshot();

    dataPref = await remark()
      .use(() => toc({ exclude: "Headline 1.1.1" }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);

    // exclude array
    data = await remark()
      .use(toc)
      .process(
        mdYamlPrefs({ exclude: "['Table of Contents', 'Headline 1.1.1']" })
      );
    expect(data.contents).toMatchSnapshot();

    dataPref = await remark()
      .use(() => toc({ exclude: ["Table of Contents", "Headline 1.1.1"] }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);
  });

  test("tight to tight toc", async () => {
    let data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ tight: true }));
    expect(data.contents).toMatchSnapshot();

    let dataPref = await remark()
      .use(() => toc({ tight: true }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);

    data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ tight: false }));
    expect(data.contents).toMatchSnapshot();

    dataPref = await remark()
      .use(() => toc({ tight: false }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);
  });

  test("ordered to ordered toc", async () => {
    let data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ ordered: true }));
    expect(data.contents).toMatchSnapshot();

    let dataPref = await remark()
      .use(() => toc({ ordered: true }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);

    data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ tight: true, ordered: true }));
    expect(data.contents).toMatchSnapshot();

    dataPref = await remark()
      .use(() => toc({ tight: true, ordered: true }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);
  });

  test("fromHeading to skip headings", async () => {
    let data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ fromHeading: 1 }));
    expect(data.contents).toMatchSnapshot();

    let dataPref = await remark()
      .use(() => toc({ fromHeading: 1 }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);

    data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ fromHeading: 2 }));
    expect(data.contents).toMatchSnapshot();

    dataPref = await remark()
      .use(() => toc({ fromHeading: 2 }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);
  });

  test("toHeading to skip headings", async () => {
    let data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ fromHeading: 1, toHeading: 2 }));
    expect(data.contents).toMatchSnapshot();

    let dataPref = await remark()
      .use(() => toc({ fromHeading: 1, toHeading: 2 }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);

    data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ fromHeading: 1, toHeading: 3 }));
    expect(data.contents).toMatchSnapshot();

    dataPref = await remark()
      .use(() => toc({ fromHeading: 1, toHeading: 3 }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);
  });

  test("set class of toc", async () => {
    let data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ className: "my-toc my-toc--mod" }));
    expect(data.contents).toMatchSnapshot();

    let dataPref = await remark()
      .use(() => toc({ className: "my-toc my-toc--mod" }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);

    // illegal class-name
    data = await remark()
      .use(toc)
      .process(mdYamlPrefs({ className: "my-toc my-toc--$" }));
    expect(data.contents).toMatchSnapshot();

    dataPref = await remark()
      .use(() => toc({ className: "my-toc my-toc--$" }))
      .process(mdDefaultPrefs());

    expect(data.contents).toEqual(dataPref.contents);
  });
});
