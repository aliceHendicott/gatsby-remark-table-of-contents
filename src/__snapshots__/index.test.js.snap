// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOC function exclude to exclude header 1`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Headline 1.1.0](#headline-110)

    *   [Headline 1.1.1](#headline-111)

        *   [Headline 1.1.1.0](#headline-1110)

*   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function exclude to exclude header 2`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Table of Contents](#table-of-contents)

*   [Headline 1.1.0](#headline-110)

    *   *   [Headline 1.1.1.0](#headline-1110)

*   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function exclude to exclude header 3`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Headline 1.1.0](#headline-110)

    *   *   [Headline 1.1.1.0](#headline-1110)

*   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function fromHeading to skip headings 1`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Headlinge 1.0.0](#headlinge-100)

    *   [Table of Contents](#table-of-contents)

    *   [Headline 1.1.0](#headline-110)

        *   [Headline 1.1.1](#headline-111)

            *   [Headline 1.1.1.0](#headline-1110)

    *   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function fromHeading to skip headings 2`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Table of Contents](#table-of-contents)

*   [Headline 1.1.0](#headline-110)

    *   [Headline 1.1.1](#headline-111)

        *   [Headline 1.1.1.0](#headline-1110)

*   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function illegal toc-definition should not throw an error but take defaults 1`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Table of Contents](#table-of-contents)

*   [Headline 1.1.0](#headline-110)

    *   [Headline 1.1.1](#headline-111)

        *   [Headline 1.1.1.0](#headline-1110)

*   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function ordered to ordered toc 1`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

1.  [Table of Contents](#table-of-contents)

2.  [Headline 1.1.0](#headline-110)

    1.  [Headline 1.1.1](#headline-111)

        1.  [Headline 1.1.1.0](#headline-1110)

3.  [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function ordered to ordered toc 2`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

1.  [Table of Contents](#table-of-contents)
2.  [Headline 1.1.0](#headline-110)
    1.  [Headline 1.1.1](#headline-111)
        1.  [Headline 1.1.1.0](#headline-1110)
3.  [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function set class of toc 1`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"my-toc my-toc--mod\\">

*   [Table of Contents](#table-of-contents)

*   [Headline 1.1.0](#headline-110)

    *   [Headline 1.1.1](#headline-111)

        *   [Headline 1.1.1.0](#headline-1110)

*   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function set class of toc 2`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Table of Contents](#table-of-contents)

*   [Headline 1.1.0](#headline-110)

    *   [Headline 1.1.1](#headline-111)

        *   [Headline 1.1.1.0](#headline-1110)

*   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function set type of toc to mdx 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 18,
              "line": 3,
              "offset": 19,
            },
            "start": Object {
              "column": 3,
              "line": 3,
              "offset": 4,
            },
          },
          "type": "text",
          "value": "Headlinge 1.0.0",
        },
      ],
      "depth": 1,
      "position": Object {
        "end": Object {
          "column": 18,
          "line": 3,
          "offset": 19,
        },
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 2,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 21,
              "line": 5,
              "offset": 41,
            },
            "start": Object {
              "column": 4,
              "line": 5,
              "offset": 24,
            },
          },
          "type": "text",
          "value": "Table of Contents",
        },
      ],
      "depth": 2,
      "position": Object {
        "end": Object {
          "column": 21,
          "line": 5,
          "offset": 41,
        },
        "start": Object {
          "column": 1,
          "line": 5,
          "offset": 21,
        },
      },
      "type": "heading",
    },
    Object {
      "type": "jsx",
      "value": "<div className=\\"toc\\">",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "Table of Contents",
                    },
                  ],
                  "title": null,
                  "type": "link",
                  "url": "#table-of-contents",
                },
              ],
              "type": "paragraph",
            },
          ],
          "spread": false,
          "type": "listItem",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "Headline 1.1.0",
                    },
                  ],
                  "title": null,
                  "type": "link",
                  "url": "#headline-110",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "type": "text",
                              "value": "Headline 1.1.1",
                            },
                          ],
                          "title": null,
                          "type": "link",
                          "url": "#headline-111",
                        },
                      ],
                      "type": "paragraph",
                    },
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "children": Array [
                                Object {
                                  "children": Array [
                                    Object {
                                      "type": "text",
                                      "value": "Headline 1.1.1.0",
                                    },
                                  ],
                                  "title": null,
                                  "type": "link",
                                  "url": "#headline-1110",
                                },
                              ],
                              "type": "paragraph",
                            },
                          ],
                          "spread": false,
                          "type": "listItem",
                        },
                      ],
                      "ordered": false,
                      "spread": false,
                      "type": "list",
                    },
                  ],
                  "spread": true,
                  "type": "listItem",
                },
              ],
              "ordered": false,
              "spread": true,
              "type": "list",
            },
          ],
          "spread": true,
          "type": "listItem",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "Headline 1.2.0",
                    },
                  ],
                  "title": null,
                  "type": "link",
                  "url": "#headline-120",
                },
              ],
              "type": "paragraph",
            },
          ],
          "spread": false,
          "type": "listItem",
        },
      ],
      "ordered": false,
      "spread": true,
      "type": "list",
    },
    Object {
      "type": "jsx",
      "value": "</div>",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 18,
              "line": 13,
              "offset": 119,
            },
            "start": Object {
              "column": 4,
              "line": 13,
              "offset": 105,
            },
          },
          "type": "text",
          "value": "Headline 1.1.0",
        },
      ],
      "depth": 2,
      "position": Object {
        "end": Object {
          "column": 18,
          "line": 13,
          "offset": 119,
        },
        "start": Object {
          "column": 1,
          "line": 13,
          "offset": 102,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 19,
              "line": 15,
              "offset": 139,
            },
            "start": Object {
              "column": 5,
              "line": 15,
              "offset": 125,
            },
          },
          "type": "text",
          "value": "Headline 1.1.1",
        },
      ],
      "depth": 3,
      "position": Object {
        "end": Object {
          "column": 19,
          "line": 15,
          "offset": 139,
        },
        "start": Object {
          "column": 1,
          "line": 15,
          "offset": 121,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 22,
              "line": 16,
              "offset": 161,
            },
            "start": Object {
              "column": 6,
              "line": 16,
              "offset": 145,
            },
          },
          "type": "text",
          "value": "Headline 1.1.1.0",
        },
      ],
      "depth": 4,
      "position": Object {
        "end": Object {
          "column": 22,
          "line": 16,
          "offset": 161,
        },
        "start": Object {
          "column": 1,
          "line": 16,
          "offset": 140,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 18,
              "line": 18,
              "offset": 180,
            },
            "start": Object {
              "column": 4,
              "line": 18,
              "offset": 166,
            },
          },
          "type": "text",
          "value": "Headline 1.2.0",
        },
      ],
      "depth": 2,
      "position": Object {
        "end": Object {
          "column": 18,
          "line": 18,
          "offset": 180,
        },
        "start": Object {
          "column": 1,
          "line": 18,
          "offset": 163,
        },
      },
      "type": "heading",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 20,
      "offset": 182,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`TOC function set type of toc to not to mdx 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 18,
              "line": 3,
              "offset": 19,
            },
            "start": Object {
              "column": 3,
              "line": 3,
              "offset": 4,
            },
          },
          "type": "text",
          "value": "Headlinge 1.0.0",
        },
      ],
      "depth": 1,
      "position": Object {
        "end": Object {
          "column": 18,
          "line": 3,
          "offset": 19,
        },
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 2,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 21,
              "line": 5,
              "offset": 41,
            },
            "start": Object {
              "column": 4,
              "line": 5,
              "offset": 24,
            },
          },
          "type": "text",
          "value": "Table of Contents",
        },
      ],
      "depth": 2,
      "position": Object {
        "end": Object {
          "column": 21,
          "line": 5,
          "offset": 41,
        },
        "start": Object {
          "column": 1,
          "line": 5,
          "offset": 21,
        },
      },
      "type": "heading",
    },
    Object {
      "type": "html",
      "value": "<div class=\\"toc\\">",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "Table of Contents",
                    },
                  ],
                  "title": null,
                  "type": "link",
                  "url": "#table-of-contents",
                },
              ],
              "type": "paragraph",
            },
          ],
          "spread": false,
          "type": "listItem",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "Headline 1.1.0",
                    },
                  ],
                  "title": null,
                  "type": "link",
                  "url": "#headline-110",
                },
              ],
              "type": "paragraph",
            },
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "type": "text",
                              "value": "Headline 1.1.1",
                            },
                          ],
                          "title": null,
                          "type": "link",
                          "url": "#headline-111",
                        },
                      ],
                      "type": "paragraph",
                    },
                    Object {
                      "children": Array [
                        Object {
                          "children": Array [
                            Object {
                              "children": Array [
                                Object {
                                  "children": Array [
                                    Object {
                                      "type": "text",
                                      "value": "Headline 1.1.1.0",
                                    },
                                  ],
                                  "title": null,
                                  "type": "link",
                                  "url": "#headline-1110",
                                },
                              ],
                              "type": "paragraph",
                            },
                          ],
                          "spread": false,
                          "type": "listItem",
                        },
                      ],
                      "ordered": false,
                      "spread": false,
                      "type": "list",
                    },
                  ],
                  "spread": true,
                  "type": "listItem",
                },
              ],
              "ordered": false,
              "spread": true,
              "type": "list",
            },
          ],
          "spread": true,
          "type": "listItem",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [
                    Object {
                      "type": "text",
                      "value": "Headline 1.2.0",
                    },
                  ],
                  "title": null,
                  "type": "link",
                  "url": "#headline-120",
                },
              ],
              "type": "paragraph",
            },
          ],
          "spread": false,
          "type": "listItem",
        },
      ],
      "ordered": false,
      "spread": true,
      "type": "list",
    },
    Object {
      "type": "html",
      "value": "</div>",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 18,
              "line": 13,
              "offset": 119,
            },
            "start": Object {
              "column": 4,
              "line": 13,
              "offset": 105,
            },
          },
          "type": "text",
          "value": "Headline 1.1.0",
        },
      ],
      "depth": 2,
      "position": Object {
        "end": Object {
          "column": 18,
          "line": 13,
          "offset": 119,
        },
        "start": Object {
          "column": 1,
          "line": 13,
          "offset": 102,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 19,
              "line": 15,
              "offset": 139,
            },
            "start": Object {
              "column": 5,
              "line": 15,
              "offset": 125,
            },
          },
          "type": "text",
          "value": "Headline 1.1.1",
        },
      ],
      "depth": 3,
      "position": Object {
        "end": Object {
          "column": 19,
          "line": 15,
          "offset": 139,
        },
        "start": Object {
          "column": 1,
          "line": 15,
          "offset": 121,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 22,
              "line": 16,
              "offset": 161,
            },
            "start": Object {
              "column": 6,
              "line": 16,
              "offset": 145,
            },
          },
          "type": "text",
          "value": "Headline 1.1.1.0",
        },
      ],
      "depth": 4,
      "position": Object {
        "end": Object {
          "column": 22,
          "line": 16,
          "offset": 161,
        },
        "start": Object {
          "column": 1,
          "line": 16,
          "offset": 140,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 18,
              "line": 18,
              "offset": 180,
            },
            "start": Object {
              "column": 4,
              "line": 18,
              "offset": 166,
            },
          },
          "type": "text",
          "value": "Headline 1.2.0",
        },
      ],
      "depth": 2,
      "position": Object {
        "end": Object {
          "column": 18,
          "line": 18,
          "offset": 180,
        },
        "start": Object {
          "column": 1,
          "line": 18,
          "offset": 163,
        },
      },
      "type": "heading",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 20,
      "offset": 182,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`TOC function tight to tight toc 1`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Table of Contents](#table-of-contents)
*   [Headline 1.1.0](#headline-110)
    *   [Headline 1.1.1](#headline-111)
        *   [Headline 1.1.1.0](#headline-1110)
*   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function tight to tight toc 2`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Table of Contents](#table-of-contents)

*   [Headline 1.1.0](#headline-110)

    *   [Headline 1.1.1](#headline-111)

        *   [Headline 1.1.1.0](#headline-1110)

*   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function toHeading to skip headings 1`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Headlinge 1.0.0](#headlinge-100)

    *   [Table of Contents](#table-of-contents)
    *   [Headline 1.1.0](#headline-110)
    *   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;

exports[`TOC function toHeading to skip headings 2`] = `
"# Headlinge 1.0.0

## Table of Contents

<div class=\\"toc\\">

*   [Headlinge 1.0.0](#headlinge-100)

    *   [Table of Contents](#table-of-contents)

    *   [Headline 1.1.0](#headline-110)

        *   [Headline 1.1.1](#headline-111)

    *   [Headline 1.2.0](#headline-120)

</div>

## Headline 1.1.0

### Headline 1.1.1

#### Headline 1.1.1.0

## Headline 1.2.0
"
`;
